src/util/find_uid.c

```diff
@@ -324,10 +329,88 @@ static errno_t get_active_uid_linux(hash_table_t *table, uid_t search_uid)
     }
     return ret;
 }
+#endif
 
-errno_t get_uid_table(TALLOC_CTX *mem_ctx, hash_table_t **table)
+#ifdef __FreeBSD__
+static errno_t get_active_uid_freebsd(hash_table_t *table, uid_t uid)
+{
+    struct kinfo_proc *kp;
+    hash_key_t key;
+    hash_value_t value;
+    size_t sz;
+    int err, mib[3];
+
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PROC;
+
+    sz = 0;
+    err = sysctl(mib, 3, NULL, &sz, NULL, 0);
+    if (err != 0) {
+        err = errno;
+        DEBUG(SSSDBG_CRIT_FAILURE, "sysctl failed.\n");
+        return err;
+    }
+    sz *= 2;
+
+    kp = talloc_size(NULL, sz);
+    if (kp == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "talloc failed.\n");
+        return ENOMEM;
+    }
+
+    err = sysctl(mib, 3, kp, &sz, NULL, 0);
+    if (err) {
+        err = errno;
+        DEBUG(SSSDBG_CRIT_FAILURE, "sysctl failed.\n");
+        talloc_free(kp);
+        return err;
+    }
+
+    err = (table != NULL ? EOK : ENOENT);
+    for (size_t i = 0; i < sz / sizeof(struct kinfo_proc); i++) {
+        if (kp[i].ki_pid == 0) {
+            continue;
+        }
+
+        if (table != NULL) {
+            key.type = HASH_KEY_ULONG;
+            key.ul = (unsigned long) kp[i].ki_ruid;
+            value.type = HASH_VALUE_ULONG;
+            value.ul = (unsigned long) kp[i].ki_ruid;
+
+            err = hash_enter(table, &key, &value);
+            if (err != HASH_SUCCESS) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
+                      "cannot add to table [%s]\n", hash_error_string(err));
+                err = ENOMEM;
+                break;
+            }
+        } else {
+            if (kp[i].ki_ruid == uid) {
+                err = EOK;
+                break;
+            }
+        }
+    }
+    talloc_free(kp);
+    return err;
```

Comment:
Since these two sentences seem to be the most common ones before returning I would add a `done:` label to jump and wrap up everything before returning